File: python_api/examples/servo_move_to.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""
servo_mode_to.py

Пример отправки в сервопривод команд для перемещения в требуемую позицию.

"""

import Servo
import time
import bus_handler

#Порт шины данных. По умолчпнию /dev/RS_485
port        = '/dev/RS_485'

#Адрес сервопривода. По умолчанию 10
servo_id    = 10

#Инициализация шины передачи данных
master = bus_handler.Bus(port = port, baudrate = 460800, debug = False, timeout = 1.0)

#Инициализация сервопривода
servo = Servo.Servo(servo_id, master.bus)

#Включение питания обмоток мотора
servo.set_torque(1)

if __name__ == '__main__':

    while(True):
        
        #Движение в позицию 4000        
        servo.set_point(4000)

        #Ожидание перемещения мотора до требуемой позиции
        time.sleep(4)

        servo.set_point(0)
        time.sleep(4)

        servo.set_point(-4000)
        time.sleep(4)

        servo.set_point(0)
        time.sleep(4)   

File: c_api/examples/servo_move_to.c
/******************************************************************************
     * File: servo_move_to.c
     * Description: Пример работы с сервоприводом. Перемещение в требуемую позицию.
******************************************************************************/
#include "Servo.h"
#include "bus_handler.h"

//Адрес сервопривода. По умолчанию 10.
int servo_addr = 10;

int main()
{

  //Инициализация последовательного порта для работы с сервоприводом. По умолчанию /dev/RS_485
  if (init_port("/dev/RS_485",false) < 0)
  {
    return -1;
  }

  //Инициализация сервопривода
  init_servo(servo_addr);

  //Стандартный ре-им работы сервопривода
  set_PID_mode(servo_addr,NORMAL);

  //Включение питания обмоток двигателя
  set_torque(servo_addr,true);
  while (1) 
  {  

    //Перемещение на позицию 4000
    set_point(servo_addr,4000);
    sleep(4);

    set_point(servo_addr,0);
    sleep(4);

    set_point(servo_addr,-4000);
    sleep(4);   

  }   
  return 0; 
}